python/doc/source/numpy.rst

```diff
@@ -0,0 +1,77 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. _numpy_interop:
+
+Using PyArrow with NumPy
+========================
+
+PyArrow allows converting back and forth from
+`NumPy <https://www.numpy.org/>`_ arrays to Arrow :ref:`Arrays <data.array>`.
+
+NumPy to Arrow
+--------------
+
+To convert a NumPy array to Arrow, one can simply call the :func:`pyarrow.array`
+factory function.  This function will always make a copy of the input data,
+as Arrow has stricter alignment requirements than NumPy.
+
+.. code-block:: pycon
```

Comment:
Use ipython directive here so the output will be rendered by the latest version of the library

Reply:
It doesn't highlight the prompts, which "pycon" (short for Python console) does.

Reply:
I see. Unless I am misunderstanding is not very maintainable at scale; I don't think we should get into the habit of copy-pasting console output into the documentation.  My book has over 1000 such generated code examples; I can't imagine having to maintain these outputs without auto-generating them. I think we would be better investing our time adding appropriate highlighting to the output of the ipython directive